File: benchmarks/vm/09_struct_loop.rb
Acc = Struct.new(:a, :b)

t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
acc = Acc.new(1, 2)
200_000.times do |i|
  acc.a = acc.a + i
  acc.b = acc.b + acc.a
end
total = acc.a + acc.b
total_ms = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - t0) * 1000
puts "total = #{total}"
puts format("BENCH_RESULT: %.3f ms", total_ms)

File: benchmarks/vm/09_struct_loop.lua
local t0 = os.clock()
local acc = { a = 1, b = 2 }
for i = 0, 199999 do
    acc.a = acc.a + i
    acc.b = acc.b + acc.a
end
local total = acc.a + acc.b
local total_ms = (os.clock() - t0) * 1000
print("total = " .. tostring(total))
print(string.format("BENCH_RESULT: %.3f ms", total_ms))
